storage_image/models/storage_relation_abstract.py

```diff
@@ -16,7 +16,7 @@ class ImageRelationAbstract(models.AbstractModel):
     _order = "sequence, image_id"
 
     sequence = fields.Integer()
-    image_id = fields.Many2one("storage.image", required=True)
+    image_id = fields.Many2one("storage.image", required=True, ondelete="cascade")
```

Comment:
I'm not sure this is a good idea every time. Maybe we want this only from the product form? 
All in all you could potentially use an image for many records. Probably this `ondelete` has been skipped on purpose.
I would override `unlink` and use a ctx key to control the behavior.
Then in the product form, set this ctx key. Or, it can be configuration on the storage.
Alternatively you could use `restrict` and show a btn to manage images for the current record and from there you could delete the images 1st.

I know it sounds complicated but maybe users won't expect to delete images right away.
@lmignon @sebastienbeau what's your POV?

Reply:
I agree with @simahawk ...

|  Alternatively you could use restrict and show a btn to manage images for the current record and from there you could delete the images 1st.

:+1: 

An other motivation to not `unlink` an image is to avoid dead link into html pages where the image is referenced..

Reply:
You are right, I read it the other way around indeed :sweat:

Reply:
@lmignon ping

Reply:
Thank you for the extra informations @ivantodorovich. Indeed, I was wrong on the potential effect of such a cascade delete